src/debugger/launch.ts

```diff
@@ -169,13 +169,113 @@ export function createSnippetConfiguration(
+        request: "launch",
+        sourceLanguages: ["swift"],
+        name: `Test ${ctx.swiftPackage.name}`,
+        cwd: folder,
+        preLaunchTask: `swift: Build All${nameSuffix}`,
+    };
+
+    if (process.platform === "darwin") {
+        const swiftFolderPath = ctx.workspaceContext.toolchain.swiftFolderPath;
+        if (swiftFolderPath === undefined) {
+            return null;
+        }
+
+        const libraryPath = ctx.workspaceContext.toolchain.swiftTestingLibraryPath();
+        const frameworkPath = ctx.workspaceContext.toolchain.swiftTestingFrameworkPath();
+        const sanitizer = ctx.workspaceContext.toolchain.sanitizer(configuration.sanitizer);
+
+        return {
+            ...baseConfig,
+            program: path.join(
+                buildDirectory,
+                "debug",
+                `${ctx.swiftPackage.name}PackageTests.swift-testing`
+            ),
+            args: ["--experimental-event-stream-output", fifoPipePath],
+            env: {
+                ...testEnv,
+                ...sanitizer?.runtimeEnvironment,
+                DYLD_FRAMEWORK_PATH: frameworkPath,
+                DYLD_LIBRARY_PATH: libraryPath,
+                SWT_SF_SYMBOLS_ENABLED: "0",
+            },
+        };
+    } else if (process.platform === "win32") {
+        // On Windows, add XCTest.dll to the Path
+        // and run the .xctest executable from the .build directory.
+        const runtimePath = ctx.workspaceContext.toolchain.runtimePath;
+        const xcTestPath = ctx.workspaceContext.toolchain.xcTestPath;
+        if (xcTestPath === undefined) {
+            return null;
+        }
+        if (xcTestPath !== runtimePath) {
+            testEnv.Path = `${xcTestPath};${testEnv.Path ?? process.env.Path}`;
+        }
+        const sdkroot = configuration.sdk === "" ? process.env.SDKROOT : configuration.sdk;
+        if (sdkroot === undefined) {
+            return null;
+        }
+        let preRunCommands: string[] | undefined;
+        if (
+            configuration.debugger.useDebugAdapterFromToolchain ||
+            vscode.workspace.getConfiguration("lldb")?.get<string>("library")
+        ) {
+            preRunCommands = [`settings set target.sdk-path ${sdkroot}`];
+        }
+
+        return {
+            ...baseConfig,
+            program: path.join(
+                buildDirectory,
+                "debug",
+                `${ctx.swiftPackage.name}PackageTests.swift-testing`
+            ),
+            args: ["--experimental-event-stream-output", fifoPipePath],
+            env: testEnv,
+            preRunCommands: preRunCommands,
+        };
+    } else {
+        // On Linux, just run the .xctest executable from the configured build directory.
+        return {
+            ...baseConfig,
+            program: path.join(
+                buildDirectory,
+                "debug",
+                ctx.swiftPackage.name + "PackageTests.xctest"
+            ),
+            env: testEnv,
```

Comment:
This should be `swift-testing`. Also `--experimental-event-stream-output` isn't setup.

Reply:
Fixed